docs/dev/connectors/cassandra.md

```diff
@@ -78,76 +96,195 @@ Note that that enabling this feature will have an adverse impact on latency.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+// get the execution environment
+final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+// get input data by connecting to the socket
+DataStream<String> text = env.socketTextStream(hostname, port, "\n");
+
+// parse the data, group it, window it, and aggregate the counts
+DataStream<Tuple2<String, Long>> result = text
+
+        .flatMap(new FlatMapFunction<String, Tuple2<String, Long>>() {
+            @Override
+            public void flatMap(String value, Collector<Tuple2<String, Long>> out) {
+                // normalize and split the line
+                String[] words = value.toLowerCase().split("\\W+");
+
+                // emit the pairs
+                for (String word : words) {
+                    //Do not accept empty word, since word is defined as primary key in C* table
+                    if (!word.isEmpty()) {
+                        out.collect(new Tuple2<String, Long>(word, 1L));
+                    }
+                }
+            }
+        })
+
+        .keyBy(0)
+        .timeWindow(Time.seconds(5))
+        .sum(1)
+        ;
+
+CassandraSink.addSink(result)
+        .setQuery("INSERT INTO example.wordcount(word, count) values (?, ?);")
+        .setHost("127.0.0.1")
+        .build();
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+CassandraSink.addSink(input.javaStream)
+    .setQuery("INSERT INTO test.wordcount(word, count) values (?, ?);")
+    .setClusterBuilder(new ClusterBuilder() {
+        @Override
+        def buildCluster(builder: Cluster.Builder): Cluster = {
+            builder.addContactPoint("127.0.0.1").build()
+        }
+    })
+    .build()
 {% endhighlight %}
 </div>
 </div>
 
-The Cassandra sinks support both tuples and POJO's that use DataStax annotations.
-Flink automatically detects which type of input is used.
 
-Example for such a Pojo:
+### Cassandra Sink Example for Streaming POJO Data Type
+An example of streaming a POJO data type and store the same POJO entity back to Cassandra. In addition, this POJO implementation needs to follow [DataStax Java Driver Manual](http://docs.datastax.com/en/developer/java-driver/2.1/manual/object_mapper/creating/) to annotate the class as Cassandra connector internally maps each field of this entity to an associated column of the desginated Table using `com.datastax.driver.mapping.Mapper` class of DataStax Java Driver.
+
+Please note that if the upsert query was set, an `IllegalArgumentException` would be thrown with the following error message `Specifying a query is not allowed when using a Pojo-Stream as input.`
+
+For each CQL defined data type for columns, please refer to [CQL Documentation](https://docs.datastax.com/en/cql/3.1/cql/cql_reference/cql_data_types_c.html)
```

Comment:
the first part of the sentence doesn't quite make sense to me, the wording is a bit odd.

Reply:
Will rephrase. Thank you.